api/types/github.go

```diff
@@ -161,6 +161,12 @@ func (c *GithubConnectorV3) CheckAndSetDefaults() error {
 	if err := c.Metadata.CheckAndSetDefaults(); err != nil {
 		return trace.Wrap(err)
 	}
+	if c.Version == "" {
+		c.Version = V3
+	}
+	if c.Kind == "" {
+		c.Kind = KindGithubConnector
+	}
```

Comment:
Shouldn't a similar default be provided for all the resources?

Reply:
I'm going to make a PR for that and other validation consistency, likely today. Old draft - #5958

Reply:
Yeah, let's do that separately.
For this PR I want to keep the defaulting behavior as close as possible to what JSON schema did.